### DISCUSSION TITLE
Schema.prisma is not reading the environment variables injected by elastic beanstalk

### DISCUSSION THREAD

**Discussion by NathanFrank285** [prisma/prisma#15561] [Q&A] [answered]
### Bug description

Hello, when I set the env variable DATABASE_URL in elastic beanstalk, then SSH into the deployment and run 'npx prisma migrate deploy' I get the error:

`[ec2-user@ip-172-31-26-198 current]$ npx prisma migrate deploy
Prisma schema loaded from prisma/schema.prisma
Error: Schema validation error - Error (query-engine-node-api library)
Error code: P1012
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:10
   | 
 9 |   provider = "postgresql"
10 |   url      = env("DATABASE_URL")
   | 

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 4.3.1`

if I hardcode in the url, the migration will work, but i'd like for it to be dynamic as we'll have multiple environments. It appears that prisma searches the app for a .env file to draw the values from when this is run

### Expected behavior

The schema should find the DATABASE_URL being added by elastic beanstalk and the database migrations should run.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

```ts

```

### Environment & setup

- OS: macOS Monterey version 12.4
- Database: PostgreSQL 13.7
- Node.js version: 16.17

### Prisma Version

```
prisma                  : 4.3.1
@prisma/client          : 4.3.1
Current platform        : rhel-openssl-1.0.x
Query Engine (Node-API) : libquery-engine c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/libquery_engine-rhel-openssl-1.0.x.so.node)
Migration Engine        : migration-engine-cli c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/migration-engine-rhel-openssl-1.0.x)
Introspection Engine    : introspection-core c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/introspection-engine-rhel-openssl-1.0.x)
Format Binary           : prisma-fmt c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/prisma-fmt-rhel-openssl-1.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.3.0-32.c875e43600dfe042452e0b868f7a48b817b9640b
Default Engines Hash    : c875e43600dfe042452e0b868f7a48b817b9640b
Studio                  : 0.473.0
```

**Comment by jkomyno** [comment, marked as answer]
Hi @NathanFrank285, thanks for opening this issue.

> It appears that prisma searches the app for a .env file to draw the values from when this is run

Prisma resorts to the `.env` file only if it cannot find the variables it needs in the environment. 
As a sanity check, can you please print your database url by running `echo $DATABASE_URL` before attempting to run `npx prisma migrate deploy` in your `[ec2-user@ip-172-31-26-198 current]` SSH session? I think the issue is that the environment variable set in Elastic Beanstalk is not available in your SSH session, as it is not persisted to [your profile file](https://unix.stackexchange.com/questions/117467/how-to-permanently-set-environmental-variables).

**Reply by NathanFrank285** [reply]
Hey @jkomyno, Thanks for getting back to me. Ah yes you are 100% spot on here. When I run the echo command the response is empty, however when I us the eb cli to printenv the variable is there, confirming as you said the env variable is not available in the ssh session. Would you recommend that I use an aws procfile to run ```npx prisma migrate deploy``` post-deploy or something else? 
I didn't see any specific resources for EB in the docs, but do you happen to have any guides for best practices when deploying to elastic beanstalk?
Thanks!!

**Reply by NathanFrank285** [reply]
Wanted to update, I added ```npx prisma migrate deploy``` into a .ebextension file to run on every deploy and it is migrating the db! Thank you

**Reply by nmn28** [reply]
hi, i am having a similar issue: 

```
npx prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MongoDB database

Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:10
   | 
 9 |   provider = "mongodb"
10 |   url      = env("DATABASE_URL")
   | 

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 5.2.0
```

i then used `echo $DATABASE_URL`

and tried the `npx prisma db push` again only to get:
```
npx prisma db push
Need to install the following packages:
  prisma@5.2.0
Ok to proceed? (y) y
Error: Could not find a schema.prisma file that is required for this command.
You can either provide it with --schema, set it as `prisma.schema` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
```
please note i have a schema.prisma file in my folder

please let me know if you have advice

**Reply by enosefelix** [reply]
Yeah, you can fix this by adding the path to your schema.prisma, example:
npx prisma db push --schema=<path to your prisma.schema file>

**Reply by Sharifur** [reply]
yes, it worked thanks.

**Comment by simplyStyle** [comment]
There is another solution, which is to set dotenv

```js
import dotenv from "dotenv";

dotenv.config();
```

**Reply by krishna6371** [reply]
which file you write

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
